apps/iotune/iotune.cc

```diff
@@ -349,6 +362,15 @@ class io_worker {
             } else {
                 _rates.push_back(_requests);
             }
+
+            // Drop samples that got measured during the warm up period.
+            // We drop an extra sample. This is because the one second timer
+            // logic is really independent of when we start counting requests in
+            // `issue_request` itself so the first bucket might not be a full
+            // measurement and cause skew otherwise.
+            size_t samples_to_drop = warmup_period(_duration) / period + 1;
+            _rates.erase(_rates.begin(), _rates.begin() + std::min(samples_to_drop, _rates.size()));
```

Comment:
Rates here are one number per shard, that is number of requests executed by one worker in duration time, they have no relation with the period. I don't think you have anything else to do here than adjust the start and end measurement points in io_worker.

Reply:
Ah, wait, it's generic rate collector, not specific to sharded :facepalm: